stdlib/path.gr

```diff
@@ -597,7 +597,7 @@ provide let relativeTo = (source, dest) => {
   let pathInfo2 = pathInfo(dest)
   let (base2, _, _) = pathInfo2
   match ((base1, base2)) {
-    (Abs(_), Rel(_)) | (Abs(_), Rel(_)) => Err(Incompatible(DifferentBases)),
+    (Abs(_), Rel(_)) | (Rel(_), Abs(_)) => Err(Incompatible(DifferentBases)),
```

Comment:
We might wanna split this fix into a seperate pr and add a regression test.

Reply:
https://github.com/grain-lang/grain/pull/2054